DeepCat.js

```diff
@@ -209,7 +208,24 @@
 		newSearchTerms = DeepCat.computeResponses( responses, newSearchTerms );
 		newSearchTerms = DeepCat.computeErrors( errors, newSearchTerms );
 
-		substituteSearchRequest( newSearchTerms.join( ' ' ) );
+		logAndFinishRequest( newSearchTerms.join( ' ' ) );
+	}
+
+	function logAndFinishRequest( newSearchTermString ) {
+		var logRequestUrl = '//tools.wmflabs.org/catgraph-jsonp/logrequestlength?searchquerylength={0}';
+		return $.ajax( {
+			url: stringFormat( logRequestUrl, newSearchTermString.length ),
+			timeout: ajaxTimeout,
+			cache: false,
+			complete: function() {
+				finishDeepCatRequest( newSearchTermString );
+			}
+		} );
+	}
+
+	function finishDeepCatRequest( newSearchTermString ) {
+		substituteSearchRequest( newSearchTermString );
+		removeAjaxThrobber();
```

Comment:
Yeah true and fully understandable. I wished there was another way. I tested a few other things again, but with everything else I tried we get less then 1/4 of the logs. Submitting the page seems to stop every other process there. :-/